stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/group/member/GroupChatInfoMemberOptionsViewModel.kt

```diff
@@ -0,0 +1,130 @@
+    chatClient: ChatClient = ChatClient.instance(),
+) : ViewModel() {
+
+    private val channelClient: ChannelClient = chatClient.channel(cid)
+    private val _events = MutableLiveData<Event<UiEvent>>()
+    private val _state: MediatorLiveData<State> = MediatorLiveData()
+    val events: LiveData<Event<UiEvent>> = _events
+    val state: LiveData<State> = _state
+
+    init {
+        viewModelScope.launch {
+            val result = chatDomain.useCases.queryChannels(
+                filter = Filters.and(
+                    Filters.eq("type", "messaging"),
+                    FilterObject(
+                        data = mutableMapOf(
+                            "distinct" to true,
+                            "members" to listOf(memberId, chatDomain.currentUser.id),
+                        ),
+                    ),
+                ),
+                sort = QuerySort.desc("last_updated"),
+                messageLimit = 0,
+                limit = 1,
+            ).await()
+            if (result.isSuccess) {
+                _state.addSource(
+                    Transformations.map(result.data().channelsState) { channelState ->
+                        when (channelState) {
+                            is QueryChannelsController.ChannelsState.Result -> {
+                                State(
+                                    directChannelCid = channelState.channels.filterNot { channel ->
+                                        channel.isDraft
+                                    }.firstOrNull()?.cid,
+                                    loading = false,
+                                )
+                            }
+                            QueryChannelsController.ChannelsState.NoQueryActive,
+                            QueryChannelsController.ChannelsState.Loading,
+                            -> State(directChannelCid = null, loading = true)
+                            QueryChannelsController.ChannelsState.OfflineNoResults -> State(
+                                directChannelCid = null,
+                                loading = false,
+                            )
+                        }
+                    }
+                ) { newState -> _state.value = newState }
+            }
+        }
+    }
+
+    fun onAction(action: Action) {
+        when (action) {
+            Action.MessageClicked -> handleMessageClicked()
+            Action.RemoveFromChannel -> removeFromChannel()
+        }
+    }
+
+    private fun handleMessageClicked() {
+        val state = state.value!!
+        _events.value = Event(
+            if (state.directChannelCid != null) {
+                UiEvent.RedirectToChat(state.directChannelCid)
+            } else {
+                UiEvent.RedirectToChatPreview
+            },
+        )
+    }
+
+    private fun removeFromChannel() {
+        viewModelScope.launch {
+            val result = channelClient.removeMembers(memberId).await()
```

Comment:
We need to add offline SDK support for this use case. If it's not a part of this PR, then it's good to create a new task and put it in our backlog

Reply:
https://stream-io.atlassian.net/browse/CAS-648